### PR TITLE
Feature/litpop unc helper

### PR DESCRIPTION
**Pull request by chahank**
This pull request makes some updates to the uncertainty quantification modules.

Main changes:

- update of the plotting methods
- addition of a choice of exposure for the exposure and entity helper methods (allows e.g., to use different litpop exponents).
- addresses issue #323

Question: Should the module be renamed from `uncertainty_quantification` to something like `uncsens_quantification` or else in order to make it more explicit that it contains different methods to assess the robustness of the model?

### CODE REVIEW

File: climada/engine/uncertainty_quantification/calc_base.py [hunk 1]
```diff
@@ -65,10 +65,10 @@ def check_distr(self):
                 if input_param_name in distr_dict:
                     func = distr_dict[input_param_name]
                     x = np.linspace(func.ppf(0.01), func.ppf(0.99), 100)
-                    if not np.all(func.pdf(x) == input_param_func.pdf(x)):
+                    if not np.all(func.cdf(x) == input_param_func.cdf(x)):
```
Comment (chahank) [inline climada/engine/uncertainty_quantification/calc_base.py:68]: The pdf does not exist for discrete distribution, the cdf however is always defined.

File: climada/engine/uncertainty_quantification/unc_output.py [hunk 2]
```diff
@@ -367,7 +373,7 @@ def get_largest_si(self, salib_si, metric_list=None):
 
         #get max index
         si_df_num = si_df.select_dtypes('number')
-        si_df_num[si_df_num<0.001] = 0 #remove noise whenn all si are 0
+        si_df_num[si_df_num<0.01] = 0 #remove noise whenn all si are 0
```
Comment (emanuel-schmid) [inline climada/engine/uncertainty_quantification/unc_output.py:376]:
```suggestion
        si_df_num[si_df_num<threshold] = 0 #remove noise whenn all si are 0
```

File: climada/engine/uncertainty_quantification/unc_output.py [hunk 3]
```diff
@@ -609,26 +620,40 @@ def plot_rp_uncertainty(self, figsize=(16, 6), axes=None):
...
             ax.fill_between(losses, count + 2*n, 2*n)
+            if add_orig:
+                ax.plot(
+                    [orig_list[n], orig_list[n]], [2*n, 2*(n+1)],
+                    color=colors[n], linestyle='dotted', linewidth=2,
+                    label="orig=%.2f%s" %u_vtm(orig_list[n])
+                    );
 
         ax.set_xlim(min_l, max_l)
         ax.set_ylim(0, 2*(n+1))
```
Comment (emanuel-schmid) [inline climada/engine/uncertainty_quantification/unc_output.py:651]: n may be undefined, I suggest to:
```
        ax.set_ylim(0, 2*(unc_df.shape[1]))
```

File: climada/engine/uncertainty_quantification/unc_output.py [hunk 4]
```diff
@@ -609,26 +620,40 @@ def plot_rp_uncertainty(self, figsize=(16, 6), axes=None):
...
-        ax.set_xlabel('impact')
-        ax.set_ylabel('return period [years]')
+        ax.set_xlabel('Impact [%s]' %self.unit)
+        ax.set_ylabel('Return period [years]')
         ax.set_yticks(np.arange(0, 2*(n+1), 2))
```
Comment (emanuel-schmid) [inline climada/engine/uncertainty_quantification/unc_output.py:654]:
```
        ax.set_yticks(np.arange(0, 2*(unc_df.shape[1]), 2))
```
s.a.

File: climada/engine/uncertainty_quantification/input_var.py [hunk 5]
```diff
@@ -446,29 +461,31 @@ def ent(impf_set, disc_rate, exp, meas_set, haz_id_dict= None,
...
             The base discount rates.
-        exp : climada.entity.exposures.base.Exposure
-            The base exposure.
+        exp_list : [climada.entity.exposures.base.Exposure]
+            The list of base exposure. Can be one or many to uniformly sample
+            from.
         meas_set : climada.entity.measures.measure_set.MeasureSet
             The base measures.
-        haz_id_dict : dict(), optional
+        haz_id_dict : dict()
```
Comment (emanuel-schmid) [inline climada/engine/uncertainty_quantification/input_var.py:476]:
```suggestion
        haz_id_dict : dict
```

File: climada/engine/uncertainty_quantification/input_var.py [hunk 6]
```diff
@@ -560,22 +583,23 @@ def entfut(impf_set, exp, meas_set, haz_id_dict=None,
...
             The base impact function set.
-        exp : climada.entity.exposures.base.Exposure
-            The base exposure.
+        exp_list : [climada.entity.exposures.base.Exposure]
+            The list of base exposure. Can be one or many to uniformly sample
+            from.
         meas_set : climada.entity.measures.measure_set.MeasureSet
             The base measures.
-        haz_id_dict : dict(), optional
+        haz_id_dict : dict()
```
Comment (emanuel-schmid) [inline climada/engine/uncertainty_quantification/input_var.py:591]:
```suggestion
        haz_id_dict : dict
```

File: climada/engine/uncertainty_quantification/input_var.py [hunk 7]
```diff
@@ -284,13 +287,17 @@ def exp(exp, bounds_totval=None, bounds_noise=None):
...
+            From the provided list of exposure is elements are uniformly
+            sampled. For example, LitPop instances with different exponents.
 
         If a bounds is None, this parameter is assumed to have no uncertainty.
 
         Parameters
         ----------
-        exp : climada.entity.exposures.Exposures
-            The base exposure.
+        exp_list : [climada.entity.exposures.Exposures]
```
Comment (emanuel-schmid) [inline climada/engine/uncertainty_quantification/input_var.py:298]:
```suggestion
        exp_list : list of climada.entity.exposures.Exposures
```

**Comment by chahank** [comment]
The `n` to `unc_df.shape[1]` was done in e9e1602

**Comment by chahank** [comment]
Thanks! All changes were accepted and commited.

**Pull request merged**